### PR TITLE
Add reducers, actions and client functions for user access tokens

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Add reducers, actions and client functions for user access tokens.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6987

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by jwilander** [comment]
Waiting on server PR to merge, I'll request reviews when it goes in

File: src/actions/users.js [hunk 1]
```diff
@@ -1122,6 +1122,153 @@ export function switchLdapToEmail(ldapPassword, email, emailPassword, mfaCode =
...
+export function createUserAccessToken(userId, description) {
+    return async (dispatch, getState) => {
+        dispatch({type: UserTypes.CREATE_USER_ACCESS_TOKEN_REQUEST});
+
+        let data;
+        try {
+            data = await Client4.createUserAccessToken(userId, description);
+        } catch (error) {
+            dispatch({type: UserTypes.CREATE_USER_ACCESS_TOKEN_FAILURE, error});
+            return {error};
```
Comment (enahum) [inline src/actions/users.js]: there's no need for checking the `forceLogoutIfNecessary` or logErrors in here and in the other actions?

File: src/utils/user_utils.js [hunk 2]
```diff
@@ -33,20 +33,37 @@ export function displayUsername(user, teammateNameDisplay) {
     return name;
 }
 
+export function rolesIncludePermission(roles, permission) {
+    const rolesArray = roles.split(' ');
+    return rolesArray.indexOf(permission) !== -1;
```
Comment (enahum) [inline src/utils/user_utils.js]: maybe you can do `return rolesArray.includes(permission)`  and that will return a boolean

Comment (jwilander) [inline src/utils/user_utils.js]: Didn't know that existed, will do

File: src/actions/users.js (hunk 1 above)
Comment (jwilander) [inline src/actions/users.js]: Don't we want to do that on all actions? If we get a 401 back we want to log out the user right?

Comment (jwilander) [inline src/actions/users.js]: Misread, will update

**Comment by jwilander** [comment]
@enahum changes made

**Pull request merged**